### PR TITLE
etc-lite-patch <- v1.12.2

### PR DESCRIPTION
**Pull request by meowsbits**
Merges upstream (ethereum/go-ethereum) thru latest v1.12.2 release into the `etc-lite-patch` branch.

### CODE REVIEW

File: cmd/geth/main.go [hunk 1]
```diff
@@ -442,7 +442,8 @@ func startNode(ctx *cli.Context, stack *node.Node, backend ethapi.Backend, isCon
 		}
 		// Set the gas price to the limits from the CLI and start mining
 		gasprice := flags.GlobalBig(ctx, utils.MinerGasPriceFlag.Name)
-		ethBackend.TxPool().SetGasPrice(gasprice)
+		// ethBackend.TxPool().SetGasPrice(gasprice)
+		ethBackend.TxPool().SetGasTip(gasprice)
```
Comment (ziogaschr) [inline cmd/geth/main.go]: Duplicate line, one has to be removed

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -568,11 +568,6 @@ var (
 		Value:    ethconfig.Defaults.Miner.Recommit,
 		Category: flags.MinerCategory,
 	}
-	MinerNoVerifyFlag = &cli.BoolFlag{
```
Comment (ziogaschr) [inline cmd/utils/flags.go]: What if we keep this flag as well for ethash?

File: core/txpool/subpool.go [hunk 3]
```diff
@@ -104,7 +104,7 @@ type SubPool interface {
 	// Add enqueues a batch of transactions into the pool if they are valid. Due
 	// to the large transaction churn, add may postpone fully integrating the tx
 	// to a later point to batch multiple ones together.
-	Add(txs []*Transaction, local bool, sync bool) []error
```
Comment (ziogaschr) [inline core/txpool/subpool.go]: Want we break support of blobTXs (and bloppool) if we change the type from `txpool.Transaction` to `types.Transaction`?

Txpool at this point wants to just be a wrapper of its subpools, while `Locals` is a notion of the legacypool and pre-merge chains.

Comment (ziogaschr) [reply core/txpool/subpool.go]: You figured it out in this commit https://github.com/etclabscore/ethereum.go-ethereum/pull/18/commits/3f44042b6ca4e418d5cdeb5e4b3200e9cd24127b already. :)

File: cmd/utils/flags.go (hunk 2 above)
Comment (meowsbits) [reply cmd/utils/flags.go]: SGTM, I removed it in a flurry.

Comment (meowsbits) [reply cmd/utils/flags.go]: [2ddfa04](https://github.com/etclabscore/ethereum.go-ethereum/pull/18/commits/2ddfa042b2de815273061ba254390e4eda4029fe) re-installs it.

**Pull request merged**